source/include/VirtualInstrument.h

```diff
@@ -140,11 +145,10 @@ class VIClump : public FunctionClump
     VirtualInstrument*  TopVI()      {
         VIClump *caller = this;
         do {
-            while (caller->_waitingClumps)
-                caller = caller->_waitingClumps;
-            while (caller->_caller)
+            if (caller->_caller)
```

Comment:
Add curly braces for the if code block, so the intent is clear and we don't rely on indentation along to convey the meaning.